lstests/doc/README.md

```diff
@@ -0,0 +1,692 @@
+
+For a writer, if `buffered=True`, it will be possible to keep information organized and make the log result human-readable. If `buffered=False`, logging will take place as soon as a line is ready. Multiple threads may log in an interleaving manner, unaware of others' existence. In general, setting `buffered=False` will *not* reduce the entire logging time, but will keep the lines in a strict chronological order, though the information itself may not be human-readable.
+
+Note that, `buffered=True` only provides a possibility to buffer. Buffering does not happen automatically. A line of information will only be buffered when it is explicitly told  to do so.
+
+For example, given two threads, both scheduled to execute
+
+```python
+# clus is a Cluster object
+clus.log("{}: first half.".format(thread_name), level="debug", buffer=True)
+clus.log("{}: second half.".format(thread_name), level="debug", buffer=True)
+clus.flush()
+```
+
+or equivalently
+
+```python
+# clus is a Cluster object
+clus.log("{}: first half.".format(thread_name), level="debug", buffer=True)
+clus.log("{}: second half.".format(thread_name), level="debug")
+```
+
+setting `buffered=True` will guarantee that the log file will either look like
+
+```
+thread-1: first half
+thread-1: second half
+thread-2: first half
+thread-2: second half
+```
+
+or
+
+```
+thread-2: first half
+thread-2: second half
+thread-1: first half
+thread-1: second half
+```
+
+but will never be of the following form
+
+```
+thread-2: first half
+thread-1: first half
+thread-1: second half
+thread-2: second half
+```
+
+#### Log Levels
+
+| level   | value |
+| ------- | ----- |
+| `OFF`   | 100   |
+| `FLAG`  | 90    |
+| `FATAL` | 60    |
+| `ERROR` | 50    |
+| `WARN`  | 40    |
+| `INFO`  | 30    |
+| `DEBUG` | 20    |
+| `TRACE` | 10    |
+| `ALL`   | 0     |
+
+The `LogLevel` class accepts either a numeric value or a case-insensitive string for the level. While there are text names given for the level values, all the integers in `[0, 100]` are valid log levels.
+
+```python
+clus.log("Some warning information.", level="warn")
+```
+
+is equivalent to
+
+```python
+clus.log("Some warning information.", level=40)
+```
+
+or
+
+```python
+clus.warn("Some warning information.")
+```
+
+As another example,
+
+```python
+clus.log("Some info more prominent than INFO but not yet a WARN", level=35)
+```
+
+This line will be written to any logging destination with a threshold at or lower than 35. It will be printed out on the screen if the `ScreenWriter` has an `INFO`-level threshold. It will be written to a log file if there is a file writer `FileWriter(threshold=35)`. But it will not appear on the log file that captures only `WARN`-or-higher-level information.
+
+### `Service`
+
+A `Service` object represents the connection with the launcher service running in the background. Once a `Service` object is created, it will try to connect the launcher service automatically.
```

Comment:
"connect the launcher" -> "connect to the launcher"

Reply:
Updated. Thank you.